Homework10/GenericArray.cs

```diff
@@ -0,0 +1,51 @@
+﻿namespace Homework10
+{
+    public class GenericArray<T> where T : Human
+    {
+        private T[] _array;
+
+        public GenericArray(int size)
+        {
+            _array = new T[size + 1];
+        }
+
+        public int GetAraryLength() => _array.Length;
+
+        public void AddElement(int index, T element)
+        {
+            _array[index] = element;
+        }
+
+        public T GetElement(int index) => _array[index];
+
+        public void RemoveElement(ref T[] _array, int index)
+        {
+            T[] tmp = new T[_array.Length - 1];
+            for (int i = 0; i < index; i++)
```

Comment:
Yes, your solution with deleting element works, but you can avoid using 2 loops. Approach with one loop:

            T[] temp2 = new T[_array.Length - 1];
            var j = 0;
            for (int i = 0; i < temp2.Length; i++)
            {
                if (i == index)
                {
                    continue;
                }

                temp2[j] = _array[i];
                j++;
            }

Reply:
updated